pkg/config/types/tls.go

```diff
@@ -0,0 +1,63 @@
+package types
+
+import (
+	"crypto/tls"
+	"fmt"
+	"sort"
+)
+
+func TLSMinVersion(minVersion string) (uint16, error) {
+	if minVersion == "" {
+		return 0, nil
+	}
+	v, ok := versions[minVersion]
+	if !ok {
+		return 0, fmt.Errorf("unsupported tls minimal version: %s supported versions:%v", minVersion, versionNames)
+	}
+	return v, nil
+}
+
+func TLSCiphers(ciphers []string) ([]uint16, error) {
+	if len(ciphers) == 0 {
+		return nil, nil
+	}
+	var res []uint16
+	for _, cipher := range ciphers {
+		v, ok := secureCiphers[cipher]
+		if !ok {
+			v, ok = insecureCiphers[cipher]
+			if !ok {
+				return nil, fmt.Errorf("unsupported tls cipher: %s supported ciphers insecure:%v, secure:%v", cipher, insecureCiphersNames, secureCiphersNames)
+			}
+		}
+		res = append(res, v)
+	}
+	return res, nil
+}
+
+var versions = map[string]uint16{
+	"TLS12": tls.VersionTLS12,
```

Comment:
we're probably going to have a similar option for setting [TLS version on dp](https://github.com/kumahq/kuma/issues/4622) should we have the same string format as Envoy (meaning `TLSv1_2` and `TLSv1_3` so that's consistent)? Or are we going to map `TLS12` to `TLSv1_2` in dp?

Reply:
What's the point of the setting on the DP? After all setting the CP forces the DP to obey no? 

As for the spelling there's really no standard anywhere around this string so makes sense following what envoy does.

Reply:
We're not setting this in there https://github.com/kumahq/kuma/blob/a3b8d2e1a802269563db45bc295b21978cb10249/pkg/plugins/runtime/gateway/filter_chain_generator.go#L245. The https API has nothing to do with the xDS API right?

Reply:
This PR adds the change to all 3 servers: `dp-server (xds), api-server and kds-server` right?

As for Envoy itself I don't know why this is set but in envoy 1.22 the default on both cleint and server is 1.2 already see: https://www.envoyproxy.io/docs/envoy/v1.22.0/api-v3/extensions/transport_sockets/tls/v3/common.proto

Reply:
> This PR adds the change to all 3 servers: dp-server (xds), api-server and kds-server right?

Maybe, I did not see any changes to `xds` packages so I thought this might be only API server. If you're sure it's for xds as well then 👍  (we should see it somewhere in golden files right?)

Also, meant to send this:
https://github.com/kumahq/kuma/blob/6cb5de9681657ce89c3a802c7a4da3cf611c322e/pkg/xds/bootstrap/template_v3.go#L190 

I just want to make sure that we're setting it everywhere :)

Reply:
> golden files
Envoy is never an XDS server right?

Anyway I've checked it it works on the XDS server (go-control-plane).

>I just want to make sure that we're setting it everywhere :)

But we don't need to set it right as it's the default for Envoy?

Reply:
I'm guessing if someone wants to override the default. But I've just read that there are issues with setting 1.3 on the client - https://github.com/envoyproxy/envoy/issues/9300 not sure if this is still true.